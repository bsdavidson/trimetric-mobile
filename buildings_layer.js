import React, {Component} from "react";
import {StyleSheet, View} from "react-native";
import {connect} from "react-redux";

import Mapbox from "@mapbox/react-native-mapbox-gl";

export function BuildingsLayer(props) {
  if (!props.layerVisibility) {
    return null;
  }
  return (
    <Mapbox.VectorSource>
      <Mapbox.FillExtrusionLayer
        id="building3d"
        sourceLayerID="building"
        style={[
          mapStyles.building,
          {
            visibility: props.layerVisibility.buildings ? "visible" : "none"
          }
        ]}
      />
    </Mapbox.VectorSource>
  );
}

function mapStateToProps(state) {
  return {
    layerVisibility: state.layerVisibility
  };
}

export default connect(mapStateToProps)(BuildingsLayer);

const mapStyles = Mapbox.StyleSheet.create({
  building: {
    fillExtrusionOpacity: 0.5,
    fillExtrusionHeight: Mapbox.StyleSheet.identity("height"),
    fillExtrusionBase: Mapbox.StyleSheet.identity("min_height"),
    fillExtrusionColor: "#FFFFFF"
  }
});
