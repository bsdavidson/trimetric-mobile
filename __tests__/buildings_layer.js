import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import {BuildingsLayer} from "../buildings_layer";

jest.mock("@mapbox/react-native-mapbox-gl");

it("renders buildingsLayer", () => {
  // render null
  const blNull = renderer.create(<BuildingsLayer />).toJSON();
  expect(blNull).toMatchSnapshot("null_buildingsLayer");

  // render non-visible buildings
  let bl = renderer
    .create(<BuildingsLayer layerVisibility={{buildings: false}} />)
    .toJSON();
  expect(bl).toMatchSnapshot("non_visible_buildingsLayer");

  // render visible buildings
  bl = renderer
    .create(<BuildingsLayer layerVisibility={{buildings: true}} />)
    .toJSON();
  expect(bl).toMatchSnapshot("visible_buildingsLayer");
});
