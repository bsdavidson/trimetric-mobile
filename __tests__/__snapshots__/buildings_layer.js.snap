// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`non_visible_buildingsLayer 1`] = `
<VectorSource>
  <FillExtrusionLayer
    id="building3d"
    sourceLayerID="building"
    style={
      Array [
        Object {
          "fillExtrusionBase": 123,
          "fillExtrusionColor": "#FFFFFF",
          "fillExtrusionHeight": 123,
          "fillExtrusionOpacity": 0.5,
        },
        Object {
          "visibility": "none",
        },
      ]
    }
  />
</VectorSource>
`;

exports[`null_buildingsLayer 1`] = `null`;

exports[`visible_buildingsLayer 1`] = `
<VectorSource>
  <FillExtrusionLayer
    id="building3d"
    sourceLayerID="building"
    style={
      Array [
        Object {
          "fillExtrusionBase": 123,
          "fillExtrusionColor": "#FFFFFF",
          "fillExtrusionHeight": 123,
          "fillExtrusionOpacity": 0.5,
        },
        Object {
          "visibility": "visible",
        },
      ]
    }
  />
</VectorSource>
`;
